File: gpt-3.5-ft-large/stage2/parallel_construct_vector_length_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:parallel construct vector_length clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    if(){
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    int lengthless1 = 0;
    int device_num = 0;
    size_t device_num_length = 1;

    int* a = (int *)malloc(n * sizeof(int));
    int** has_equal;

    has_equal = (int **)malloc(device_num_length * sizeof(int *));
    for (int x = 0; x < device_num_length; ++x){
        has_equal[x] = (int *)malloc(10 * sizeof(int));
        for (int y = 0; y < 10; ++y){
            has_equal[x][y] = 0;
        }
    }

    for (int x = 0; x < n; ++x){
        a[x] = 0;
    }

    #pragma acc enter data copyin(has_equal[0:device_num_length][0:10])
    for (int x = 0; x < device_num_length; ++x){
        #pragma acc parallel
        {
            #pragma acc loop vector
            for (int y = 0; y < 10; ++y){
                has_equal[x][y] = 1;
            }
        }
    }
    #pragma acc exit data copyout(has_equal[0:device_num_length][0:10])


    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        device_num = x;
        for (int y = 0; y < device_num_length; ++y){
            for (int z = 0; z < 10; ++z){
                has_equal[y][z] = 0;
            }
        }
    }

    failed = failed + lengthless1;
  }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/stage2/parallel_construct_vector_length_clause.f90
#ifndef T1
!T1:parallel construct vector_length clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(8),DIMENSION, ALLOCATABLE :: a, b, c
        INTEGER :: x

        ALLOCATE(a(10), b(10), c(10))

        a = 0
        b = 0

        !$acc data copyin(a(1:10), b(1:10)) copyout(c(1:10))
          !$acc parallel vector_length(4)
            !$acc loop vector
            DO x = 1, 10
              a(x) = x
            END DO
            !$acc loop vector
            DO x = 1, 10
              b(x) = x
            END DO
            !$acc loop vector_length(2) seq
            DO x = 1, 10
              c(x) = a(x) + b(x)
            END DO
          !$acc end parallel
        !$acc end data

        DO x = 1, 10
          IF (c(x) .ne. 2 * x) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM